lib/localparticipant.js

```diff
@@ -288,12 +294,13 @@ LocalParticipant.prototype.addTracks = function addTracks(tracks) {
     throw new E.INVALID_TYPE('tracks',
       'Array of LocalAudioTrack, LocalVideoTrack, LocalDataTrack, or MediaStreamTrack');
   }
-
-  var self = this;
-  return tracks.reduce(function(addedTracks, track) {
-    var addedTrack = self.addTrack(track);
-    return addedTrack ? addedTracks.concat(addedTrack) : addedTracks;
-  }, []);
+  tracks = tracks.filter(function(track) {
+    return !this.tracks.has(track.id);
+  }, this);
+  this.publishTracks(tracks).catch(function() {
+    // Do nothing.
+  });
+  return tracks;
```

Comment:
@markandrus This should be `return tracks.map(track => this.tracks.get(track.id));`

Reply:
Good catch!